### ISSUE TITLE
Make code more modular

### ISSUE THREAD

**Issue by lysol** [lysol/typeto.me#18]
Had some ideas this morning.
- Create an npm module for the backend code, as some sort of wrapper around socket.io channels.
- In that vein, have the npm module serve up the frontend JS the same way socket.io does for its client JS.
- Have the client JS provide methods for subscribing to channels/rooms/chats/whatever you want to call it.
- And wrap that in another level of JS that gives you the chat box markup and such, something more easily styled than the page is right now.

Long story short, I'd like to be able to provide an npm module that can allow people to embed this style of chat widget wherever. I think.

Or maybe nobody would really find that useful.
